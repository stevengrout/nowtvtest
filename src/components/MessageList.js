import React from 'react';
import PropTypes from 'prop-types';
import MessageComponent from "./Message"


const getMessageComponents = (messages) => {
  return messages ?
    messages.map(message =>
      <MessageComponent key={message.id} message={message.message}
                        email={message.email} avatar={message.avatar} time={message.timestamp}/>) : []
}

const MessageListComponent = (props) => {
  return (
    <div className="messageList">
      {getMessageComponents(props.messages)}
    </div>
  );
};

MessageListComponent.propTypes = {
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      message: PropTypes.string.isRequired,
      email: PropTypes.string,
      avatar: PropTypes.string,
      time: PropTypes.string
    })
  )
};



export default MessageListComponent;
