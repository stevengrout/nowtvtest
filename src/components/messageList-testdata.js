export const testData = [
  {
    "avatar": null,
    "email": "abakera@craigslist.org",
    "firstName": "Amanda",
    "id": "73f30d93-f87b-45ea-bda5-60fa4b3716b5",
    "ip": "82.87.241.19",
    "lastName": "Baker",
    "message": "Nulla tempus. Vivamus in felis eu sapien cursus vestibulum. Proin eu mi.",
    "timestamp": "2016-02-24T00:01:28Z",
    "userId": "73f30d93-f87b-45ea-bda5-60fa4b3716b5"
  },
  {
    "avatar": "http://dummyimage.com/100x100.jpg/dddddd/000000",
    "email": "awilliams8@indiegogo.com",
    "firstName": "Aaron",
    "id": "c980934a-f54e-459e-8137-0ec3654cdafb",
    "ip": "88.59.162.204",
    "lastName": "Williams",
    "message": "Quisque ut erat. Curabitur gravida nisi at nibh.",
    "timestamp": "2016-02-26T10:05:17Z",
    "userId": "c980934a-f54e-459e-8137-0ec3654cdafb"
  }
]
