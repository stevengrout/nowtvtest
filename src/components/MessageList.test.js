/* global describe it expect */

import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15.4';
import Enzyme, { shallow } from 'enzyme';
import MessageList from './MessageList';
import { testData } from "./messageList-testdata"

Enzyme.configure({ adapter: new Adapter() });

describe('MessageList details', () => {
  it('should render a list of messages', () => {
    const output = shallow((
      <MessageList messages={testData}/>
    ));
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render when list of messages is empty', () => {
    const output = shallow((
      <MessageList messages={[]}/>
    ));
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
