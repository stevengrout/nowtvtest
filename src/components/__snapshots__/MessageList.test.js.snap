exports[`MessageList details should render a list of messages 1`] = `
<div
  className="messageList">
  <MessageComponent
    avatar={null}
    email="abakera@craigslist.org"
    key="73f30d93-f87b-45ea-bda5-60fa4b3716b5"
    message="Nulla tempus. Vivamus in felis eu sapien cursus vestibulum. Proin eu mi."
    time="2016-02-24T00:01:28Z" />
  <MessageComponent
    avatar="http://dummyimage.com/100x100.jpg/dddddd/000000"
    email="awilliams8@indiegogo.com"
    key="c980934a-f54e-459e-8137-0ec3654cdafb"
    message="Quisque ut erat. Curabitur gravida nisi at nibh."
    time="2016-02-26T10:05:17Z" />
</div>
`;

exports[`MessageList details should render when list of messages is empty 1`] = `
<div
  className="messageList" />
`;
